java-components/cli/src/main/java/io/github/redhatappstudio/jvmbuild/cli/builds/BuildDiagnosticCommand.java

```diff
@@ -0,0 +1,122 @@
+                        theBuild = n;
+                        break;
+                    }
+                }
+            }
+        } else if (artifact != null) {
+            if (gav != null) {
+                throwUnspecified();
+            }
+            ArtifactBuild ab = ArtifactBuildCompleter.createNames().get(artifact);
+            theBuild = BuildConverter.buildToArtifact(client, ab);
+        } else if (gav != null) {
+            ArtifactBuild ab = GavCompleter.createNames().get(gav);
+            theBuild = BuildConverter.buildToArtifact(client, ab);
+        } else {
+            throw new RuntimeException("Must specify one of -b, -a or -g");
+        }
+        if (theBuild == null) {
+            throw new RuntimeException("Build not found");
+        }
+        int lastSlash = theBuild.getSpec().getScm().getScmURL().lastIndexOf('/') + 1;
+        String name = theBuild.getSpec().getScm().getScmURL().substring(lastSlash).replaceFirst("\\.git.*", "");
+        System.out.println("Target directory: " + targetDirectory);
+        System.out.println(
+                "Selected build: " + theBuild.getMetadata().getName() + " of " + theBuild.getSpec().getScm().getScmURL() + ':' +
+                        theBuild.getSpec().getVersion() + '\n');
+
+        List<String> dockerFiles = theBuild.getStatus().getDiagnosticDockerFiles();
+        if ((theBuild.getStatus().getFailedBuildRecipes() == null ? 0 : theBuild.getStatus().getFailedBuildRecipes().size())
+                + 1 != dockerFiles.size()) {
+            throw new RuntimeException("Mismatch between failed/current build recipe count and number of Dockerfiles.");
+        }
+        try {
+            for (int i = 0; i < dockerFiles.size(); i++) {
+                String fileName;
+                String javaVersion;
+                String tagName;
+                if (i == dockerFiles.size() - 1) {
+                    // Final one - which is successful.
+                    javaVersion = theBuild.getStatus()
+                            .getCurrentBuildRecipe()
+                            .getJavaVersion();
+                    tagName = name + ".succeed.jdk" + javaVersion;
+                } else {
+                    javaVersion = theBuild.getStatus()
+                            .getFailedBuildRecipes()
+                            .get(i)
+                            .getJavaVersion();
+                    tagName = name + ".failed.jdk" + javaVersion;
+                }
+                fileName = "Dockerfile." + tagName;
+                System.out.println(
+                        CommandLine.Help.Ansi.AUTO
+                                .string(
+                                        """
+                                                @|green To diagnose the JDK%s build:
+
+                                                |@@|yellow podman build -f %s -t %s .
+                                                podman run -it %s|@
```

Comment:
This does not actually work quite right, it gets tagged as localhost/foo, but podman run foo does not work. This would work with docker.

```
Successfully tagged localhost/kotlinx-datetime.failed.jdk11:latest
d11f6aa29f1555324eda8f4d18f56c3d489730b77fd678f60904d39fb1d295aa
[root@6c29787e7b8f project]# podman run -it kotlinx-datetime.failed.jdk11
WARN[0000] "/" is not a shared mount, this could cause issues or missing mounts with rootless containers 
WARN[0000] Using rootless single mapping into the namespace. This might break some images. Check /etc/subuid and /etc/subgid for adding sub*ids if not using a network user 
? Please select an image: 
  ▸ registry.access.redhat.com/kotlinx-datetime.failed.jdk11:latest
    registry.redhat.io/kotlinx-datetime.failed.jdk11:latest
    docker.io/library/kotlinx-datetime.failed.jdk11:latest
```

Reply:
Strange, I tested with podman (on linux) and it worked ok - I can take another look.